src/app/Layouts/defaultPageWrapper.jsx

```diff
@@ -22,27 +72,24 @@ const PageWrapper = ({
   <DialContextProvider dials={dials}>
     <ServiceContextProvider service={service}>
       <GlobalStyle />
-      <RequestContextProvider
-        bbcOrigin={bbcOrigin}
+      <PageWithRequestContext
+        data={data}
         id={id}
         isAmp={isAmp}
+        bbcOrigin={bbcOrigin}
         pageType={pageType}
         service={service}
       >
-        <ServiceWorkerContainer />
-        <ManifestContainer />
-        <ConsentBanner />
-        <HeaderContainer />
         {children}
-        <FooterContainer />
-      </RequestContextProvider>
+      </PageWithRequestContext>
     </ServiceContextProvider>
   </DialContextProvider>
 );
 
 PageWrapper.propTypes = {
   bbcOrigin: string,
   children: node.isRequired,
+  data: shape({ any }).isRequired,
```

Comment:
Can it really be `any` shape, or do we have a structure for what the data should look like? if there are shared features maybe we should add something in `models/propTypes`?

Reply:
@oluoluoxenfree was there any update to this?

Reply:
@rhell, @jamesdonoh said the way I was trying to do it was overcomplicated, and it didn't work anyway. He said it was sufficient to check something was passed.

I was trying to do something like:

`data: oneOf({setofArticlepropTypes, frontPageSetOfProptypes]`

It was throwing other errors, and I think it's not for choosing between a set of props anyway. 

Happy for suggestions though!

Reply:
Would just something like `data: shape({ pageData: shape({ metadata: shape({ etc...` (without `isRequired`) work?

Reply:
Update: this seems not straightforward due to differences in the service JSON between news and front pages, we'll investigate and find a better prop type validation solution as a separate issue.